src/System.Linq.Expressions/src/System/Dynamic/DynamicObject.cs

```diff
@@ -229,42 +229,62 @@ public override DynamicMetaObject BindGetMember(GetMemberBinder binder)
             {
                 if (IsOverridden(nameof(DynamicObject.TryGetMember)))
                 {
-                    return CallMethodWithResult(nameof(DynamicObject.TryGetMember), binder, s_noArgs, (e) => binder.FallbackGetMember(this, e));
+                    return BindGetMemberOverridden(binder);
```

Comment:
Thanks, good point. In fact, I just noticed that `this` and `binder` are available to these helper methods anyway, so I'll revisit the code to take advantage of that and pass these to the delegate; there's no need to have a closure at all over these variables. For the remaining methods that have additional arguments, I'll go with the `local` aliasing approach.